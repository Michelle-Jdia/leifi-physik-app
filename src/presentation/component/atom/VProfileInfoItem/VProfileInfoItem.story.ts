import type { Meta, StoryFn } from '@storybook/vue3';
import VProfileInfoItem from '@/atom/VProfileInfoItem/VProfileInfoItem.vue';

export default {
    component: VProfileInfoItem,
} as Meta<typeof VProfileInfoItem>;

const Template: StoryFn<typeof VProfileInfoItem> = (args) => ({
    components: {
        VProfileInfoItem,
    },
    setup() {
        return {
            args,
        };
    },
    template: `
        <v-profile-info-item v-bind="args">Profile info item</v-profile-info-item>
    `,
});

export const Base = Template.bind({});
