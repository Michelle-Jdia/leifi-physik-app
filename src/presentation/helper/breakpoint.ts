export const breakpoints = {
    mobile: 576,
    tablet: 768,
};
