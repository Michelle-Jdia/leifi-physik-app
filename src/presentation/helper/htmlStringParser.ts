export const html =
    '<div class="rich-text-main"><div class="process_glossary"><p> </p><p>Zu vielen wichtigen Erkenntnissen über weißes Licht und seine Spektralfarben gelangte Newton mit einfachen Versuchen, die du <a href="/node/7678" target="_blank">hier</a>ausführlicher beschrieben findest.</p><h4 style="clear:both; margin-bottom:1em">Zerlegung von weißem Licht in die Spektralfarben</h4><figure role="group" class="rs_skip caption caption-drupal-entity"><div data-embed-button="media" data-entity-embed-display="view_mode:media.image_50_percent_center" data-entity-type="media" data-entity-uuid="73a22760-ff57-4a31-bac4-bc099cee2e72" class="align-right embedded-entity" data-langcode="de" data-entity-embed-display-settings="[]"><picture class=""><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/576spektralfarben_prismenspektrum_bild.svg" media="(max-width: 570px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/576spektralfarben_prismenspektrum_bild.svg" media="(max-width: 570px)" type="image/jpeg"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/992spektralfarben_prismenspektrum_bild.svg" media="(min-width: 571px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/992spektralfarben_prismenspektrum_bild.svg" media="(min-width: 571px)" type="image/jpeg"><img src="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/fa62eb7a83830f00436cca015aa94f39/1000spektralfarben_prismenspektrum_bild.svg" alt="" /></source></source></source></source></picture><span class="image-copyright font-italic small rs_skip"><i class="far fa-copyright"></i><span class="copyright-text">Joachim Herz Stiftung</span></span></div><figcaption>Abb. 1 Zerlegung von weißem Licht in Spektralfarben mit Hilfe eines Prismas</figcaption></figure><p style="text-align:justify">Wenn du weißes Licht in geeigneter Weise auf ein Prisma fallen lässt, so kannst du auf einem Schirm dahinter beobachten, dass das weiße Licht in verschiedene Farben zerlegt wurde. Auf dem Schirm ist jeweils ein Bereich zu sehen, der rot, orange, gelb, grün, cyan, blau oder violett gefärbt ist. Diese Farben nennt man die <strong>Spektralfarben</strong>.</p><p style="text-align:justify">Die Spektralfarben kannst du auch bei einem Regenbogen finden. Hier ist die Sonne die Quelle für weißes Licht und die Regentropfen sorgen für die Zerlegung.</p><h4 style="clear:both; margin-bottom:1em">Unzerlegbarkeit der Spektralfarben</h4><figure role="group" class="rs_skip caption caption-drupal-entity"><div data-embed-button="media" data-entity-embed-display="view_mode:media.image_50_percent_center" data-entity-type="media" data-entity-uuid="69987120-eeaa-483c-9032-b04e0511d4c0" class="align-right embedded-entity" data-langcode="de" data-entity-embed-display-settings="[]"><picture class=""><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/576spektralfarben_unzerlegbarkeit_bild.svg" media="(max-width: 570px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/576spektralfarben_unzerlegbarkeit_bild.svg" media="(max-width: 570px)" type="image/jpeg"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/992spektralfarben_unzerlegbarkeit_bild.svg" media="(min-width: 571px)" type="image/webp"><source srcset="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/992spektralfarben_unzerlegbarkeit_bild.svg" media="(min-width: 571px)" type="image/jpeg"><img src="https://www.app-api-dm4xeuy-bwyzh75a7hm5c.eu-2.platformsh.site/sites/default/files/images/215ed628eaebf3845102ea5ade417e85/1000spektralfarben_unzerlegbarkeit_bild.svg" alt="" /></source></source></source></source></picture><span class="image-copyright font-italic small rs_skip"><i class="far fa-copyright"></i><span class="copyright-text">Joachim Herz Stiftung</span></span></div><figcaption>Abb. 2 Unzerlegbarkeit der Spektralfarben</figcaption></figure><p style="text-align:justify">Wenn du nun eine beliebige Spektralfarbe wieder durch ein Prisma fallen lässt, so siehst du auf dem Schirm, dass sich die Farbe des Lichtes nicht wieder geändert hat. Spektralfarben lassen sich nicht weiter zerlegen. Man nennt sie auch reine Farben.</p><h4 style="clear:both; margin-bottom:1em">Nicht sichtbare Strahlung im Spektrum</h4><p style="text-align:justify">Neben den Spektralfarben enthält zum Beispiel das Sonnenlicht noch weitere Anteile, die du nicht mit bloßem Auge sehen kannst - auch dann nicht, wenn du Sonnenlicht durch ein Prisma in seine Spektralfarben zerlegst.</p><p style="text-align:justify">Ein Anteil ist <strong>infrarote Strahlung</strong>(<strong>IR-Strahlung</strong>). Diese Strahlung wäre im Experiment auf dem Schirm (vgl. <strong>Abb. 1</strong>) oberhalb des roten Lichtes zu sehen. Infrarote Strahlung wird zum Beispiel bei Wärmelampen genutzt.</p><p style="text-align:justify">Ein weiterer nicht sichtbarer Teil wird als <strong>ultraviolette Strahlung</strong>(<strong>UV-Strahlung</strong>) bezeichnet. Diese Strahlung wäre im Experiment auf dem Schirm (vgl. <strong>Abb. 1</strong>) unterhalb des blauen Lichtes zu sehen. Ultraviolette Strahlung wird zum Beispiel zum Prüfen von Geldscheinen und zum Härten von Nagellack verwendet. Auch sind Teile der UV-Strahlung für den Sonnenbrand verantwortlich.</p></div></div><div class="clearfix"></div>';

const parser = new DOMParser();

export const doc = parser.parseFromString(html, 'text/html');
