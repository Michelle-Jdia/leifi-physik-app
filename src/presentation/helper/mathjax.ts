/*
 configuration is taken from 'https://www.leifiphysik.de/themes/custom/avonis/js/mathjax-config.js?s0e9g7'
 - instead of the 'tex-mml-chtml' the 'tex-mml-svg' is used displaying the rendered functions as svg not with css & html causing errors by including fonts
 - the config for loader is removed and all the extensions are imported separately, otherwise causing errors
 */

/* eslint-disable */
// This is no longer in use. See public/script/mathjax
// export const getMathjax = async () => {
//     // @ts-ignore
//     await import('/public/script/mathjax/mathjax-config.js');
//
//     // @ts-ignore
//     await import('/public/script/mathjax/tex-mml-chtml.js');
//
//     // @ts-ignore
//     import('/public/script/mathjax/input/tex/extensions/enclose.js');
//
//     // @ts-ignore
//     import('/public/script/mathjax/input/tex/extensions/cancel.js');
//
//     // @ts-ignore
//     import('/public/script/mathjax/input/tex/extensions/color.js');
//
//     // @ts-ignore
//     // import('/public/script/mathjax/input/tex/extensions/colorv2.js');
//
//     // @ts-ignore
//     import('/public/script/mathjax/input/tex/extensions/mhchem.js');
// };
