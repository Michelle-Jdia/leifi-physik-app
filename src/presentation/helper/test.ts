import type { ComponentMountingOptions } from '@vue/test-utils';
import type { Component } from 'vue';
import { config, mount } from '@vue/test-utils';
import { expect, test } from 'vitest';

beforeAll(() => {
    config.global.renderStubDefaultSlot = true;
});

afterAll(() => {
    config.global.renderStubDefaultSlot = false;
});

export const testSnapshot = (
    component: Component,
    options?: ComponentMountingOptions<object>,
): void => {
    test('snapshot', () => {
        const wrapper = mount(component, {
            shallow: true,
            ...options,
        });

        expect(wrapper.element).toMatchSnapshot();
    });
};
