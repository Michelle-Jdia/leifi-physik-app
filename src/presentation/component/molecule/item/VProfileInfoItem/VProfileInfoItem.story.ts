import type { Meta, StoryFn } from '@storybook/vue3';
import VProfileInfoItem from '@/molecule/item/VProfileInfoItem/VProfileInfoItem.vue';

export default {
    component: VProfileInfoItem,
} as Meta<typeof VProfileInfoItem>;

const Template: StoryFn<typeof VProfileInfoItem> = (args) => ({
    components: {
        VProfileInfoItem,
    },
    setup() {
        return {
            args,
        };
    },
    template: `
        <v-profile-info-item v-bind="args">
            <template #title>Name</template>
            Mia Mustermann
        </v-profile-info-item>
    `,
});

export const Base = Template.bind({});

Base.args = {};
