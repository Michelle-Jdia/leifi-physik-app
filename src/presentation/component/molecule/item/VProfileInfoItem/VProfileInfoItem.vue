<script setup lang="ts">
import type { ClassValue } from 'clsx';
import { cn } from '@/presentation/helper/style';
import VButton from '@/atom/button/VButton/VButton.vue';

interface Props {
    modifier?: ClassValue;
}

defineProps<Props>();
</script>

<template>
    <div :class="cn('profile-info-item v-flex v-justify-between v-py-box-md', modifier)">
        <div>
            <h4 class="v-h4"><slot name="title" /></h4>

            <p
                v-if="$slots.default"
                class="v-h4 v-mt-box-sm v-inline-block v-font-fflight v-text-gray"
            >
                <slot />
            </p>
        </div>

        <v-button modifier="v-h-max v-w-max v-font-bold v-underline" fill="clear" size="none">
            Bearbeiten
        </v-button>
    </div>
</template>

<style>
ion-button {
    --background-hover: unset;
    --color: var(--ty-copy);
}
</style>
