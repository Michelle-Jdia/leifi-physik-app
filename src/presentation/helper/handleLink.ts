import {
    isDownloadStringType,
    isIssueStringType,
    isTaskStringType,
} from '@/data/type/helper/contentType';
import { getContextualRouteNameByType } from '@/connection/helper/navigation';
import router from '@/connection/router';

function clickHandler(this: HTMLAnchorElement, event: MouseEvent) {
    const uuid = this.dataset.uuid;
    const type = this.dataset.type;

    if (
        !uuid ||
        !type ||
        !(isIssueStringType(type) || isDownloadStringType(type) || isTaskStringType(type))
    ) {
        return;
    }

    event.preventDefault();

    router.push({
        name: getContextualRouteNameByType(type),
        params: {
            id: uuid,
        },
    });

    this.removeEventListener('click', clickHandler);
}

export function handleLinksClick(containerElement: HTMLElement | null): void {
    if (!containerElement) {
        return;
    }

    const links = document.getElementsByTagName('a');

    Array.from(links).forEach((link: HTMLAnchorElement) => {
        link.addEventListener('click', clickHandler);
    });
}
