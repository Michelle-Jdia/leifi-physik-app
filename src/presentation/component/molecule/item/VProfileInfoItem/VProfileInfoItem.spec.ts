import { render } from '@testing-library/vue';
import { describe, expect, test } from 'vitest';
import { testSnapshot } from '@/presentation/helper/test';
import VProfileInfoItem from '@/molecule/item/VProfileInfoItem/VProfileInfoItem.vue';

describe('VProfileInfoItem', () => {
    const options = {
        slots: {
            default: 'Mia',
        },
    };

    test('shows the correct text', () => {
        const { getByText } = render(VProfileInfoItem, options);

        getByText('Mia');
    });

    test('doesnt show the text if it was not passed', () => {
        const { queryByText } = render(VProfileInfoItem, {
            ...options,
            props: {
                value: '',
            },
            slots: {},
        });

        expect(queryByText('Mia')).toBeFalsy();
    });

    testSnapshot(VProfileInfoItem, options);
});
