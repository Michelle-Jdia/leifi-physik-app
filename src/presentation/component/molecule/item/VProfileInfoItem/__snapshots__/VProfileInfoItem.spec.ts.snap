// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VProfileInfoItem > snapshot 1`] = `
<div
  class="profile-info-item v-flex v-justify-between v-py-box-md"
>
  <div>
    <h4
      class="v-h4"
    >
      
      
    </h4>
    <p
      class="v-h4 v-mt-box-sm v-inline-block v-font-fflight v-text-gray"
    >
      
      Mia
      
    </p>
  </div>
  <v-button-stub
    fill="clear"
    isrounded="true"
    modifier="v-h-max v-w-max v-font-bold v-underline"
    size="none"
  >
     Bearbeiten 
  </v-button-stub>
</div>
`;
